### PR TITLE
#603  Default to `preserve-modules` when building packages

### PR DESCRIPTION
**Pull request by LukeSheard**
- Default `--preserve-modules` to `true` when building packages to enable `export`

field in `package.json` files.

- Fix bug where typings were generated relative to the package directory and not
the `src` directory.

- Add `--private` argument for building packages marked `private: true` in their
`package.json`

### CODE REVIEW

**Review by cangarugula — APPROVED** [review]

File: packages/modular-scripts/src/__tests__/index.test.ts [hunk 1]
```diff
@@ -204,11 +204,11 @@ describe('modular-scripts', () => {
...
           "license": "UNLICENSED",
-          "main": "dist-cjs/sample-view.cjs.js",
+          "main": "dist-cjs/index.js",
           "modular": Object {
             "type": "view",
           },
-          "module": "dist-es/sample-view.es.js",
+          "module": "dist-es/index.js",
           "name": "sample-view",
           "typings": "dist-types/src/index.d.ts",
```
Comment (cangarugula) [inline packages/modular-scripts/src/__tests__/index.test.ts:213]: https://github.com/jpmorganchase/modular/pull/608/files#diff-4870b5ce0bd4c0a47c7dfc47c42422d6e2455d523069b2f0cfbeaf1a70e759b9R33 removed the src folder so you just need update the typings path

**Review by cangarugula — CHANGES_REQUESTED** [review]

**Comment by cangarugula** [comment]
I also noticed that running builds with preserve modules adds the index.js and index.js.map files in dist-cjs but doesn't remove the current app.cjs.js and app.cjs.js.map files in the dir. It's not really an issue for production but this may seem odd when you're testing builds locally with the upgrade. We'll probably note this with the release but the upgrade won't be super tidy.

File: packages/modular-scripts/src/__tests__/index.test.ts (hunk 1 above)
Comment (LukeSheard) [reply packages/modular-scripts/src/__tests__/index.test.ts]: Resolved this and added a test case for it!

**Comment by LukeSheard** [comment]
> doesn't remove the current app.cjs.js and app.cjs.js.map files in the dir

I've added `fs.emptyDir` before package builds to match `app` build behaviour so this should be resolved now.

**Review by cangarugula — APPROVED** [review]

**Pull request merged**